src/libecalc/infrastructure/file_io.py

```diff
@@ -396,7 +400,7 @@ def _validate_headers(headers: List[str]):
                 "[ _ - # + : . , /] "
             )
         elif re.match(r"^Unnamed: \d+$", header):
-            raise ValueError("CSV input file must include header")
+            raise InvalidResourceHeaderException(f"One or more headers are missing in input-file {name}")
```

Comment:
in my opinion we should probably not have a naution about filename here. It may not always be a filename, it may be a csv dataset retrieved from another place? Therefore, a specific exception as you have here is good, and then catch that higher up, where you can add the additional information about the filename. how does that sound?

Reply:
Yes. Basically, if we have an error reading a csv file we would probably have to "stop the line", as in stop execution in order for the user to fix the error to continue? So, yes, as late as possible, higher up the hierarchy, where we are actually trying to read a file called "test.csv", or a data stream...which in theory is also possible...close to the error you have the details of what went wrong (characters on the specific line aso.), and higher up you have the details of the greater task (read a file)..

Reply:
@TeeeJay : The error is now catched differently. Please have a look and see if it is better.